scripts/firedrake-install

```diff
@@ -1266,13 +1300,24 @@ if mode == "install":
     # Force Cython to install first to work around pip dependency issues.
     run_pip_install(["Cython>=0.22"])
 
-    with environment(**compiler_env):
-        # Need to install petsc first in order to resolve hdf5 dependency.
-        if not args.honour_petsc_dir:
+    # Need to install petsc first in order to resolve hdf5 dependency.
+    if not args.honour_petsc_dir:
+        with environment(**compiler_env):
             with pipargs("--no-deps"):
                 packages.remove("petsc")
                 install("petsc/")
```

Comment:
Since we no longer `pip install` petsc, we must also switch to building slepc ourselves.

Reply:
I can't see any evidence that this is true. Why can't we just pip install slepc?

Reply:
Previously it was the case that if you had non-pip-installed petsc you couldn't pip install slepc (maybe this is no longer true).